File: prop-all/src/test/java/io/pleo/prop/objects/ComplexObjects.java
package io.pleo.prop.objects;

import javax.inject.Inject;
import javax.inject.Named;
import java.util.List;

import io.pleo.prop.core.Prop;

public class ComplexObjects {
  private Prop<InjectedObject> myComplexObjectProp;
  private Prop<InjectedObjectWithConstructor> withConstructor;
  private Prop<List<InjectedObject>> myListOfComplexObjectProp;
  private Prop<String> myStringProp;

  @Inject
  public ComplexObjects(@Named("io.pleo.test.prop1") Prop<InjectedObject> myComplexObjectProp,
                        @Named("io.pleo.test.prop1") Prop<InjectedObjectWithConstructor> withConstructor,
                        @Named("io.pleo.test.prop2") Prop<List<InjectedObject>> myListOfComplexObjectProp,
                        @Named("io.pleo.test.prop3") Prop<String> myStringProp) {
    this.myComplexObjectProp = myComplexObjectProp;
    this.withConstructor = withConstructor;
    this.myListOfComplexObjectProp = myListOfComplexObjectProp;
    this.myStringProp = myStringProp;
  }

  public Prop<InjectedObjectWithConstructor> getWithConstructor() {
    return withConstructor;
  }

  public Prop<InjectedObject> getMyComplexObjectProp() {
    return myComplexObjectProp;
  }

  public Prop<List<InjectedObject>> getMyListOfComplexObjectProp() {
    return myListOfComplexObjectProp;
  }

  public Prop<String> getMyStringProp() {
    return myStringProp;
  }
}

File: prop-all/src/test/kotlin/io/pleo/prop/objects/ComplexObjects.kt
package io.pleo.prop.objects

import com.google.inject.Inject
import com.google.inject.name.Named
import io.pleo.prop.core.Prop

@Suppress("unused")
class ComplexObjects
@Inject
constructor(
    @Named("io.pleo.test.prop1") val myComplexObjectProp: Prop<InjectedObject>,
    @Named("io.pleo.test.prop1") val withConstructor: Prop<InjectedObjectWithConstructor>,
    @Named("io.pleo.test.prop2") val myListOfComplexObjectProp: Prop<List<InjectedObject>>,
    @Named("io.pleo.test.prop3") val myStringProp: Prop<String>,
    @Named("io.pleo.test.prop7") val myMappedEnumProp: Prop<Map<ParsingStage, String>>,
) {
    enum class ParsingStage {
        UPLOADED,
        OCR,
        PENDING_HUMAN_REVIEW,
        UNDER_HUMAN_REVIEW,
        OUTPUT,
        FINISHED,
        ERROR,
    }
}
